ocaml-lsp-server/src/custom_requests/req_documentation.ml

```diff
@@ -0,0 +1,120 @@
+open Import
+module TextDocumentPositionParams = Lsp.Types.TextDocumentPositionParams
+
+let meth = "ocamllsp/documentation"
+
+let capability = ("handleDocumentation", `Bool true)
+
+module GetDocClientCapabilities = struct
+  type t = { contentFormat : MarkupKind.t list option }
+
+  let yojson_of_t { contentFormat } =
+    `Assoc
+      (match contentFormat with
+      | Some formats ->
+        [ ( "contentFormat"
+          , `List
+              (List.map
+                 ~f:(fun format -> MarkupKind.yojson_of_t format)
+                 formats) )
+        ]
+      | None -> [])
+
+  let t_of_yojson json =
+    let open Yojson.Safe.Util in
+    let contentFormat =
+      json |> member "contentFormat" |> to_list
+      |> List.map ~f:(fun format -> MarkupKind.t_of_yojson format)
+      |> Option.some
+    in
```

Comment:
Since it is an option, we should probably match if the list is empty in order to provide a `None`.
btw @voodoos I am not sure about the level of indirection, using an option on list leads to strange behaviour, ie: `Some []`, why not directly going for a list where `[] = always plaintext` ?

Reply:
> why not directly going for a list where [] = always plaintext
This makes sense. The next commit will implement it this way and we can eval if it's better